scripts/apps/search/controllers/SearchMenuController.ts

```diff
@@ -71,17 +74,18 @@ export default function SearchMenuController(
             let providers = [];
             const shortcuts = savedSearches
                 .filter((search) => search.shortcut && search.is_global)
-                .map((search) => ({
-                    _id: search._id,
-                    name: search.name,
-                    filter: search.filter,
-                }));
+                .map(mapFiltersServerToClient);
 
             // bundle repo and its shortcuts
             this.providers.forEach((provider) => {
                 providers.push(provider);
                 providers = providers.concat($filter('sortByName')(
-                    shortcuts.filter((shortcut) => get(shortcut, 'filter.query.repo', '') === provider._id),
+                    shortcuts.filter((shortcut) => {
```

Comment:
Can you explain what are you filtering here? It's not obvious from the code at all to me. What is `provider._id`? any random string or can also be a name of one of the repositories?

It would also read easier to me if you used `x == null`, `x != null` explicitly instead of `!x` and `if(x)` when your intent is to check if the value is null.

Reply:
will write it in there rather than here